### PR TITLE
Make url that's shared in a company's member invite external and form…

### PR DESCRIPTION
**Pull request by rhklee**
Fix for #972

plan: fe0d3b00-3fee-47d5-9f46-a1722b4bf873 3x

### CODE REVIEW

File: arbeitszeit_flask/templates/member/notification-about-work-invitation.html [hunk 1]
```diff
@@ -1,7 +1,3 @@
-<p>{{
-    gettext(
-        'You have been invited by a company to join them as a member on Arbeitszeitapp. Accept or reject via this link: "%(invitation_url)s".',
-        invitation_url=invitation_url
-    )
-  }}</p>
+<p>{{ gettext("You have been invited by a company to join them as a member on Arbeitszeitapp. Accept or reject via this link:") }}</p>
+<p><a href="{{ invitation_url }}">{{ invitation_url }}</a></p>
```
Comment (Amittai-Aviram) [inline arbeitszeit_flask/templates/member/notification-about-work-invitation.html:2]: Good!  But does the link text need to be the bare URL?  Could it be something like "confirm"?  I understand that this is not the goal of this PR.

**Review by Amittai-Aviram — COMMENTED** [review]
This looks great, but I guess we need to have some sort of test for it before merging with master.  Is that true?

**Comment by sloschert** [comment]
Two remarks and one proposel: 
1) We use this `get_answer_company_work_invite_url` method twice: First, when creating the email sent out to workers (in `InviteWorkerPresenterImpl`), second, when showing the work invite on the worker's dashboard (in `ShowCompanyWorkInviteDetailsPresenter`). Only in the first case this url has to be absolute.
2) How can we test if the url is absolute or relative? As it is now, we could only test this on the flask level with an integration test, because the url_index test implementation on www level (`tests/www/presenters/url_index.py`) does only know about name and arguments of the url_index methods and is converting these into a fake url. 

-> If you change the signature of the method by adding a key word argument "is_absolute", we could 1) distinguish between cases where we need an absolute url and where we don't and 2) could test that the correct argument is passed in the corresponding presenter tests.

Like this:
`def get_answer_company_work_invite_url(self, *, invite_id: UUID, is_absolute: bool) -> str:`

**Review by sloschert — CHANGES_REQUESTED** [review]
Hi, thanks for tackling this issue. I left a suggestion.

File: arbeitszeit_flask/templates/member/notification-about-work-invitation.html (hunk 1 above)
Comment (rhklee) [reply arbeitszeit_flask/templates/member/notification-about-work-invitation.html:2]: Yeah I think that's a good idea, I'll make an issue for such an enhancement for the emails.

File: tests/www/presenters/test_show_company_work_invite_details_presenter.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def test_successful_response_renders_to_view_model(self) -> None:
     def test_successful_response_renders_accept_url_to_view_model(self) -> None:
```
Comment (rhklee) [inline tests/www/presenters/test_show_company_work_invite_details_presenter.py:28]: Tests the web presenter has the properly parameterized `get_answer_company_work_invite_url` passed.

**Comment by rhklee** [comment]
I added an `is_absolute` path parameter to `get_answer_company_work_invite_url` the presenter tests now check if the `is_absolute` parameter is in the rendered views given how the UrlIndex mock is setup. I guess that seems like the easiest solution.

**Comment by rhklee** [comment]
> LGTM. Let me know if you have a plan ID for registering the consumption of this.

Cool, I just added the plan id above. And here it is again:

plan: fe0d3b00-3fee-47d5-9f46-a1722b4bf873 3x

**Comment by sloschert** [comment]
> > LGTM. Let me know if you have a plan ID for registering the consumption of this.
> 
> @seppeljordan Cool, I just added the plan id above. And here it is again:
> 
> plan: fe0d3b00-3fee-47d5-9f46-a1722b4bf873 3x

Unfortunately this plan has expired yesterday, sorry. Can you create a new one and add to the coop again?

**Comment by rhklee** [comment]
@sloschert new plan-id: `adc48617-8e5e-4cf5-8eb6-09be373428cf`

**Comment by sloschert** [comment]
Consumption registered (3x)

**Pull request merged**